src/lib.rs

```diff
@@ -841,6 +842,20 @@ impl Matches {
         }
     }
 
+    /// Returns the matching value or `None`.
+    ///
+    /// Similar to opt_default, except the two differences. Instead of
+    /// returning None when option was not present, return `def`. Instead of
+    /// returning &str slice return valued of type T parsed using str::parse().
+    pub fn opt_get<T>(&self, nm: &str, def: T)
```

Comment:
This looks like a good addition to me! What do you think about having two methods here:

```rust
pub fn opt_get<T>(&self, nm: &str) -> Result<Option<T> T::Err>;
pub fn opt_get_default<T>(&self, nm: &str, def: T) -> Result<T: T::Err>;
```

so that we can still use this method in cases where there isn't an obvious default to use?

The `Result<Option>` signature is a little weird, but chances are you'd blow up on the result, and branch on the option:

```rust
match matches.opt_get("name")? {
    Some(opt) => { ... },
    None => { ... }
}
```

Or maybe:

```rust
let opt = matches.opt_get("name").unwrap_or_else(|e| {
    println!("invalid value for `name`: {}", e);
    
    ...
    
    process::exit(1)
});
```

What do you think?

Reply:
By T::Err in the Result<> did you mean <T as FromStr>::Err ? Or is that
something new ?

> so that we can still use this method in cases where there isn't an obvious
> default to use?

In these cases won't opts_str, opt_str and opt_default be useful ? I haven't
really used getopts much, so I am not able to judge which is more ergonomic ..

Cheers,

Reply:
Ah, by `T::Err` I mean [the associated error type on the `FromStr` trait](https://doc.rust-lang.org/std/str/trait.FromStr.html#associatedtype.Err), like you have currently. We should actually be able to just say `T::Err` instead of `<T as FromStr>::Err` because all we know about `T` from the `where` clause is that it implements `FromStr`.

If you use `opt_str` you'll still need to do the `FromStr` conversion yourself, so having `opt_get` without a default is closer to how `opt_str` works already.

Reply:
Done.